src/dataframe/dataframe.jl

```diff
@@ -463,13 +496,15 @@ end
 function Base.setindex!(df::DataFrame,
                         new_df::DataFrame,
                         row_ind::Real,
-                        col_inds::AbstractVector{Bool})
+                        col_inds::AbstractVector{<:Union{Bool,Missing}})
+    length(col_inds) != ncol(df) && throw(BoundsError(df, col_inds))
     setindex!(df, new_df, row_ind, findall(col_inds))
 end
 function Base.setindex!(df::DataFrame,
                         new_df::DataFrame,
                         row_ind::Real,
                         col_inds::AbstractVector{<:ColumnIndex})
+    nrow(new_df) == 1 || throw(ArgumentError("Source DataFrame contains multiple rows"))
```

Comment:
Better say "should contain a single row" (and no initial uppercase).

Reply:
fixed